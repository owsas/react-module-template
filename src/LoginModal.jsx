import React from 'react';
import PropTypes from 'prop-types';
import { Modal, Button } from 'react-bootstrap';

const LoginModal = props => (
  <Modal show={props.show} onHide={props.onHide} {...props.modalProps} id="modalLogin">
    <Modal.Body>
      <p style={{ textAlign: 'center' }}>
      Para interactuar con este elemento primero debes iniciar sesión
      </p>
    </Modal.Body>
    <Modal.Footer>
      <Button id="modalButtonCancel" onClick={props.onHide}>Cancelar</Button>
      <Button id="modalButtonCancel" bsClass="btn btn-success" onClick={props.onLogin}>Iniciar sesión</Button>
    </Modal.Footer>
  </Modal>
);

LoginModal.defaultProps = {
  show: false,
  modalProps: {},
  onLogin: undefined,
  onHide: undefined,
};

LoginModal.propTypes = {
  show: PropTypes.bool,
  onLogin: PropTypes.func,
  onHide: PropTypes.func,
  // eslint-disable-next-line
  modalProps: PropTypes.object,
};

export default LoginModal;

