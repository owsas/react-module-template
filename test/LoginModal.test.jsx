import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import LoginModal from '../src/LoginModal';

test('should match snapshot', () => {
  const m = shallow(<LoginModal onLogin={() => {}} onHide={() => {}} />);
  expect(shallow(toJSON(m))).toMatchSnapshot();
});
