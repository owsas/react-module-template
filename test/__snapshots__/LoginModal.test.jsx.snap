// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Modal
    animation={true}
    autoFocus={true}
    backdrop={true}
    bsClass="modal"
    dialogComponentClass={[Function]}
    enforceFocus={true}
    id="modalLogin"
    keyboard={true}
    manager={
      ModalManager {
        "add": [Function],
        "containers": Array [],
        "data": Array [],
        "handleContainerOverflow": true,
        "hideSiblingNodes": true,
        "isTopModal": [Function],
        "modals": Array [],
        "remove": [Function],
      }
    }
    onHide={[Function]}
    renderBackdrop={[Function]}
    restoreFocus={true}
    show={false}
  >
    <ModalBody
      bsClass="modal-body"
      componentClass="div"
    >
      <p
        style={
          Object {
            "textAlign": "center",
          }
        }
      >
        Para interactuar con este elemento primero debes iniciar sesión
      </p>
    </ModalBody>
    <ModalFooter
      bsClass="modal-footer"
      componentClass="div"
    >
      <Button
        active={false}
        block={false}
        bsClass="btn"
        bsStyle="default"
        disabled={false}
        id="modalButtonCancel"
        onClick={[Function]}
      >
        Cancelar
      </Button>
      <Button
        active={false}
        block={false}
        bsClass="btn btn-success"
        bsStyle="default"
        disabled={false}
        id="modalButtonCancel"
        onClick={[Function]}
      >
        Iniciar sesión
      </Button>
    </ModalFooter>
  </Modal>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "animation": true,
      "autoFocus": true,
      "backdrop": true,
      "bsClass": "modal",
      "dialogComponentClass": [Function],
      "enforceFocus": true,
      "id": "modalLogin",
      "keyboard": true,
      "manager": ModalManager {
        "add": [Function],
        "containers": Array [],
        "data": Array [],
        "handleContainerOverflow": true,
        "hideSiblingNodes": true,
        "isTopModal": [Function],
        "modals": Array [],
        "remove": [Function],
      },
      "onHide": [Function],
      "renderBackdrop": [Function],
      "restoreFocus": true,
      "show": false,
    },
    "ref": undefined,
    "rendered": null,
    "type": "Modal",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "animation": true,
        "autoFocus": true,
        "backdrop": true,
        "bsClass": "modal",
        "dialogComponentClass": [Function],
        "enforceFocus": true,
        "id": "modalLogin",
        "keyboard": true,
        "manager": ModalManager {
          "add": [Function],
          "containers": Array [],
          "data": Array [],
          "handleContainerOverflow": true,
          "hideSiblingNodes": true,
          "isTopModal": [Function],
          "modals": Array [],
          "remove": [Function],
        },
        "onHide": [Function],
        "renderBackdrop": [Function],
        "restoreFocus": true,
        "show": false,
      },
      "ref": undefined,
      "rendered": null,
      "type": "Modal",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
